docs/rfc/rfc-4.adoc

```diff
@@ -0,0 +1,44 @@
+:toc: macro
+
+= RFC 4: Using the Random Beacon
+
+:icons: font
+:numbered:
+toc::[]
+
+== Overview
+
+The Random Beacon provides high-quality randomness
+which is important for secure group selection.
+
+When the Random Beacon produces an entry,
+we should use that entry immediately
+so that operators don't have too much of an opportunity
+to manipulate sortition pool state.
+Additionally,
+the beacon may be busy and unable to produce an entry at a given time.
+
+To deal with these requirements,
+we can perform group selection as follows:
+
+Alice determines that the _maintenance reward_ for wallet creation
+has grown large enough to be worth paying the cost of performing it.
+She submits a transaction calling `createNewWallet()`
+with at least a specified minimum payment.
+`createNewWallet()` calls the random beacon asking for a new entry,
+passing on the payment Alice provided,
+and specifying the callback for the random beacon.
+If the beacon is available,
+Alice is recorded as the `currentWalletCreator`
+and `createNewWallet()` is blocked from being called
+until `currentWalletCreator` is cleared.
+If the beacon is busy,
+or the provided payment is insufficient,
+the transaction reverts.
+
+Once the random beacon has generated an entry,
+it calls `selectGroup(newEntry)` with a high gas limit.
+This performs sortition from the sortition pool,
+and if successful, creates a new wallet with the selected members
+and pays the appropriate maintenance reward to Alice.
+In any case, `currentWalletCreator` is cleared.
```

Comment:
In https://github.com/keep-network/keep-core/pull/2585 it is proposed to change the beacon callback mechanism so that the given application's maintainers need to submit a transaction using an entry. If we agree on this approach, can we rework this section to say Alice is calling the second transaction and to specify if there is any expiration period for an entry delivered in the previous transaction? Also, it feels that we will have to lock the sortition pool on the first transaction from Alice, given the most recent idea from @eth-r about optimistic selection of members.